nextcord/application_command.py

```diff
@@ -756,6 +756,8 @@ def from_callback(
                         isinstance(param.annotation, type)
                         and issubclass(param.annotation, Interaction)
                     )
+                    # will always be an interaction parameter (generic with the outermost type being Interaction)
+                    or getattr(param.annotation, "__origin__", None) is Interaction
```

Comment:
Why not use [`typing.get_origin`](https://docs.python.org/3/library/typing.html#typing.get_origin)?

Reply:
Have a look at the full commit list, it was tried but this is easier on type checking

Reply:
Might work, I don't know that I've tried it yet

Reply:
Looks good to me